tests/run-with-compiler/quote-run-staged-interpreter.check

```diff
@@ -1,6 +1,4 @@
-{
-  (x: Int) => 2.+(x).+(4)
-}
+(x: Int) => 2.+(x).+(4)
```

Comment:
Could the result appear inside a "template" (class/trait)? Because then you sometimes do need parentheses around the closure to avoid `(x: Int) =>` being parsed as a self-type declaration.

Reply:
Indeed, if one writes
```scala
class Test {
  ((x: Int) => x)
}
```
It will be pretty printed as
```scala
class Test() extends Object() { 
  (x: Int) => x
}
```
which will be parsed as a self type. But I do think the lightweight syntax is worth the newly introduced bug. One can fix the issue in a separate PR